KMTF fixes on propagation coefficients (CMSSW_11_3_X)

Tyler-Lam opened the pull request:
#### PR description:

New version of KMTF algorithm as discussed in https://indico.cern.ch/event/1012620/contributions/4250430/attachments/2200192/3721090/Run3_KMTF.pdf

#### PR validation:

Nominal tests with just KMTF algorithm. No core changes, just incremental changes to KMTF

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

@rekovic The lookup tables located in L1Trigger/L1TMuon/data/bmtf_luts/kalmanLUTs.root are usually in the https://github.com/cms-l1t-offline/L1Trigger-L1TMuon.git repository.  They have to be moved but I do not have access
@dinyar please check that we didn't revert your changes

Tyler-Lam committed:
Migrated run3 KMTF changes from 11_2_X to 11_3_X

Tyler-Lam committed:
Added new kalman gains

Tyler-Lam committed:
Removed gains

Tyler-Lam committed:
Added gains for run 3 kbmtf

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33078/21389

- This PR adds an extra 252KB to repository

- Found files with invalid states:
  - L1Trigger/L1TMuon/data/bmtf_luts/kalmanLUTs.root:
    - Added: 4e78c82e3a057a792a83772489ca078e5bf8cdfd, 88f3cd16b4d3a9e0269242d4238fe497be60def6
    - Deleted: 5e66a6a1abf9e9fb3f9d2294dbdd5630ef698d93

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-checks**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33078/21389/code-checks.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33078/21389/code-checks.patch | patch -p1`
You can also run `scram build code-checks` to apply code checks directly

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33078/21389/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33078/21389/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

Tyler-Lam committed:
code checks

Tyler-Lam committed:
code-format

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33078/21402

- This PR adds an extra 256KB to repository

- Found files with invalid states:
  - L1Trigger/L1TMuon/data/bmtf_luts/kalmanLUTs.root:
    - Added: 4e78c82e3a057a792a83772489ca078e5bf8cdfd, 88f3cd16b4d3a9e0269242d4238fe497be60def6
    - Deleted: 5e66a6a1abf9e9fb3f9d2294dbdd5630ef698d93

cmsbuild commented:
A new Pull Request was created by  @Tyler-Lam  for master.

It involves the following packages:

L1Trigger/L1TMuon
L1Trigger/L1TMuonBarrel


@cmsbuild, @rekovic, @cecilecaillol can you please review it and eventually sign? Thanks.
@dinyar, @Martin-Grunewald, @thomreis this is something you requested to watch as well.
@silviodonato, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

silviodonato commented:
please test
kind reminder @cms-sw/l1-l2

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-77a5cf/13366/summary.html
**COMMIT**: f192774122cb4669092532f23c29ab1205570e2a
**CMSSW**: CMSSW_11_3_X_2021-03-08-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/33078/13366/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 38
 - DQMHistoTests: Total histograms compared: 2849195
 - DQMHistoTests: Total failures: 1695
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2847478
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 37 files compared)
 - Checked 160 log files, 37 edm output root files, 38 DQM output files

dinyar commented:
Hi @Tyler-Lam,

Thanks for the ping, looks good to me.

Cheers,
Dinyar

Tyler-Lam committed:
Scripts to make gains for high/low quality stubs

Tyler-Lam committed:
gain file

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33078/21511

- This PR adds an extra 260KB to repository

- Found files with invalid states:
  - L1Trigger/L1TMuon/data/bmtf_luts/kalmanLUTs.root:
    - Added: 4e78c82e3a057a792a83772489ca078e5bf8cdfd, 88f3cd16b4d3a9e0269242d4238fe497be60def6
    - Deleted: 5e66a6a1abf9e9fb3f9d2294dbdd5630ef698d93

cmsbuild commented:
Pull request #33078 was updated. @cmsbuild, @rekovic, @cecilecaillol can you please check and sign again.

cecilecaillol commented:
+l1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-77a5cf/13503/summary.html
**COMMIT**: c78ef02684edb3c7b0a8dc5ba33aeba71fbdf5cf
**CMSSW**: CMSSW_11_3_X_2021-03-14-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33078/13503/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2635087
 - DQMHistoTests: Total failures: 429
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2634636
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 36 files compared)
 - Checked 155 log files, 37 edm output root files, 37 DQM output files

silviodonato commented on L1Trigger/L1TMuonBarrel/src/L1TMuonBarrelKalmanAlgo.cc:
```diff
@@ -353,25 +358,41 @@ void L1TMuonBarrelKalmanAlgo::propagate(L1MuKBMTrack& track) {
+
   if (verbose_) {
-    printf("phi prop = %d* %f = %d, %d* %f =%d\n", K, aPhi_[step - 1], phi11, phiB, -bPhi_[step - 1], phi12);
+    std::cout << "phi prop = " << K << "*" << ap_fixed<BITSPARAM + 1, 2>(aPhi_[step - 1]) << " = " << phi11 << ", "
```
You should avoid using `std::cout`, please use https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMessageLogger
(see https://github.com/cms-sw/cms-sw.github.io/issues/94).
Please fix it in all the file (18 times)

Tyler-Lam replied:
Would reverting to printf() statements be ok?  We use this format to make hardware patterns that can be very expensive, so the activation of the logger would give a lot of output for no reason.

silviodonato commented on L1Trigger/L1TMuonBarrel/src/L1TMuonBarrelKalmanAlgo.cc:
```diff
@@ -353,25 +358,41 @@ void L1TMuonBarrelKalmanAlgo::propagate(L1MuKBMTrack& track) {
+  int phiBNew = ap_fixed<13, 13>(phiB11 + phiB12);
   if (verbose_) {
-    printf("phiB prop = %d* %f = %d, %d* %f =%d\n", K, aPhiB_[step - 1], phiB11, phiB, bPhiB_[step - 1], phiB12);
+    std::cout << "phiB prop = " << K << "* " << ap_fixed<BITSPARAM + 1, 2>(aPhiB_[step - 1]) << " = " << phiB11 << ", "
```
same here

silviodonato commented on L1Trigger/L1TMuonBarrel/interface/L1TMuonBarrelKalmanAlgo.h:
```diff
@@ -106,6 +106,47 @@ class L1TMuonBarrelKalmanAlgo {
+  static const int GAIN_5 = 18;
+  static const int GAIN_5INT = 0;
+
+  static const int GAIN2_0 = 18;
```
please remove the comment lines

silviodonato commented:
@Tyler-Lam @cms-sw/l1-l2
could you add a line saying explicitly that these differences are expected?

> DQMHistoTests: Total failures: 429

rekovic commented on L1Trigger/L1TMuonBarrel/test/kalmanTools/parseGainsLUT.py:
```diff
@@ -138,100 +158,105 @@ def parse(fileio,f,fout,ele,bins,mini,maxi,zeroSup=False,constant=-1,absol=False
 parse2(fileio,f,fout,0,512,0,64,True,-1,False,1)    
-parse2(fileio,f,fout,1,512,-8,8,True,-1,False,1)    
+parse2(fileio,f,fout,1,512,-8,8,True,-1,True,1)    
+#parse2(fileio,f,fout,1,512,-8,8,True,-1,False,1)    
```
remove commented line

rekovic commented on L1Trigger/L1TMuonBarrel/test/kalmanTools/parseGainsLUT.py:
```diff
@@ -51,36 +52,55 @@ def generate(tfile,name,outname,bins,mini,maxi,zeroSup=False,constant=-1,absol=F
-fileio.write('typedef ap_ufixed<9,2> updateLUTV0_t;\n')
-fileio.write('typedef ap_ufixed<9,2> updateLUTV1_t;\n');
+fileio.write('#include common.h\n')
+#fileio.write('typedef ap_ufixed<9,6> updateLUT0_t;\n')
```
remove commented lines

Tyler-Lam committed:
Removed comments

rekovic commented:
@Tyler-Lam 
We would like this backported to 11_2_X, so please address the comments.

Tyler-Lam committed:
couts removed

Tyler-Lam commented:
@rekovic I replied previously but forgot to tag @silviodonato, this emulator has to have large printouts for hardware patterns and the messagelogger output would interfere with this. I've removed the couts and reverted them to printf()

@silviodonato The emulator has been changed so we expect differences in the DQM distributions

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33078/21650

- This PR adds an extra 264KB to repository

- Found files with invalid states:
  - L1Trigger/L1TMuon/data/bmtf_luts/kalmanLUTs.root:
    - Added: 4e78c82e3a057a792a83772489ca078e5bf8cdfd, 88f3cd16b4d3a9e0269242d4238fe497be60def6
    - Deleted: 5e66a6a1abf9e9fb3f9d2294dbdd5630ef698d93

cmsbuild commented:
Pull request #33078 was updated. @cmsbuild, @rekovic, @cecilecaillol can you please check and sign again.

dinyar commented:
Hi @Tyler-Lam,

Out of interest, why are these printouts necessary for hardware patterns? Can't you just write them to a file? Reason I'm asking is that we'll probably eventually move our own pattern generators into CMSSW and I'm curious how others are tackling it.

Cheers,
Dinyar

bachtis commented:
Hi @dinyar 
What we do is have our test benches receive standard I/O. We could have it written to file but there is no reason for this or not. I prefer to stay backwards compatible 
@rekovic , KMTF was always like this I do not understand why we cannot leave our verbose flag as it was and continue to use what we have implemented . Is this OK?

silviodonato commented:
please test

rekovic commented:
Ok to sign off for L1 after passing tests.

cmsbuild commented:
-1

**Failed Tests**: RelVals RelVals-INPUT 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-77a5cf/13700/summary.html
**COMMIT**: 0936dc591f392ba23dc05a9c9dcdf9f2ca386666
**CMSSW**: CMSSW_11_3_X_2021-03-23-1100/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/33078/13700/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

- [1325.81](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-77a5cf/13700/runTheMatrix-results/1325.81_TTbar_13_106Xv1NanoAODINPUT+TTbar_13_106Xv1NanoAODINPUT+NANOEDMMC2017_106XMiniAODv1+HARVESTNANOAODMC2017_106XMiniAODv1)
```
----- Begin Fatal Exception 23-Mar-2021 17:09:49 CET-----------------------
An exception of category 'FatalRootError' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 3 stream: 0
   [1] Running path 'nanoAOD_step'
   [2] Calling method for module SimpleCandidateFlatTableProducer/'fatJetMCTable'
   Additional Info:
      [a] Fatal Root Error: @SUB=TInterpreter::AutoParse
Error parsing payload code for class reco::JetFlavourInfo with content:

#line 1 "SimDataFormatsJetMatching_xr dictionary payload"

#ifndef CMS_DICT_IMPL
  #define CMS_DICT_IMPL 1
#endif
#ifndef _REENTRANT
  #define _REENTRANT 1
#endif
#ifndef GNUSOURCE
  #define GNUSOURCE 1
#endif
#ifndef __STRICT_ANSI__
  #define __STRICT_ANSI__ 1
#endif
#ifndef GNU_GCC
  #define GNU_GCC 1
#endif
#ifndef _GNU_SOURCE
  #define _GNU_SOURCE 1
#endif
#ifndef EIGEN_DONT_PARALLELIZE
  #define EIGEN_DONT_PARALLELIZE 1
#endif
#ifndef TBB_USE_GLIBCXX_VERSION
  #define TBB_USE_GLIBCXX_VERSION 90300
#endif
#ifndef TBB_SUPPRESS_DEPRECATED_MESSAGES
  #define TBB_SUPPRESS_DEPRECATED_MESSAGES 1
#endif
#ifndef TBB_PREVIEW_RESUMABLE_TASKS
  #define TBB_PREVIEW_RESUMABLE_TASKS 1
#endif
#ifndef BOOST_SPIRIT_THREADSAFE
  #define BOOST_SPIRIT_THREADSAFE 1
#endif
#ifndef PHOENIX_THREADSAFE
  #define PHOENIX_THREADSAFE 1
#endif
#ifndef BOOST_MATH_DISABLE_STD_FPCLASSIFY
  #define BOOST_MATH_DISABLE_STD_FPCLASSIFY 1
#endif
#ifndef BOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX
  #define BOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX 1
#endif
#ifndef CMSSW_GIT_HASH
  #define CMSSW_GIT_HASH "CMSSW_11_3_X_2021-03-22-2300"
#endif
#ifndef PROJECT_NAME
  #define PROJECT_NAME "CMSSW"
#endif
#ifndef PROJECT_VERSION
  #define PROJECT_VERSION "CMSSW_11_3_X_2021-03-22-2300"
#endif
#ifndef CMSSW_REFLEX_DICT
  #define CMSSW_REFLEX_DICT 1
#endif

#define _BACKWARD_BACKWARD_WARNING_H
// Inline headers
#include <utility>
#include <vector>

#include "DataFormats/Common/interface/Wrapper.h"
#include "SimDataFormats/JetMatching/interface/JetFlavourInfo.h"
#include "SimDataFormats/JetMatching/interface/JetFlavourInfoMatching.h"
#include "SimDataFormats/JetMatching/interface/JetFlavour.h"
#include "SimDataFormats/JetMatching/interface/JetFlavourMatching.h"
#include "SimDataFormats/JetMatching/interface/MatchedPartons.h"
#include "SimDataFormats/JetMatching/interface/JetMatchedPartons.h"

#undef  _BACKWARD_BACKWARD_WARNING_H


----- End Fatal Exception -------------------------------------------------
```
- [10024.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-77a5cf/13700/runTheMatrix-results/10024.0_TTbar_13+2017+TTbar_13TeV_TuneCUETP8M1_GenSim+Digi+RecoFakeHLT+HARVESTFakeHLT+ALCA+Nano)
```
----- Begin Fatal Exception 23-Mar-2021 17:26:33 CET-----------------------
An exception of category 'FatalRootError' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'nanoAOD_step'
   [2] Calling method for module SimpleCandidateFlatTableProducer/'fatJetMCTable'
   Additional Info:
      [a] Fatal Root Error: @SUB=TInterpreter::AutoParse
Error parsing payload code for class reco::JetFlavourInfo with content:

#line 1 "SimDataFormatsJetMatching_xr dictionary payload"

#ifndef CMS_DICT_IMPL
  #define CMS_DICT_IMPL 1
#endif
#ifndef _REENTRANT
  #define _REENTRANT 1
#endif
#ifndef GNUSOURCE
  #define GNUSOURCE 1
#endif
#ifndef __STRICT_ANSI__
  #define __STRICT_ANSI__ 1
#endif
#ifndef GNU_GCC
  #define GNU_GCC 1
#endif
#ifndef _GNU_SOURCE
  #define _GNU_SOURCE 1
#endif
#ifndef EIGEN_DONT_PARALLELIZE
  #define EIGEN_DONT_PARALLELIZE 1
#endif
#ifndef TBB_USE_GLIBCXX_VERSION
  #define TBB_USE_GLIBCXX_VERSION 90300
#endif
#ifndef TBB_SUPPRESS_DEPRECATED_MESSAGES
  #define TBB_SUPPRESS_DEPRECATED_MESSAGES 1
#endif
#ifndef TBB_PREVIEW_RESUMABLE_TASKS
  #define TBB_PREVIEW_RESUMABLE_TASKS 1
#endif
#ifndef BOOST_SPIRIT_THREADSAFE
  #define BOOST_SPIRIT_THREADSAFE 1
#endif
#ifndef PHOENIX_THREADSAFE
  #define PHOENIX_THREADSAFE 1
#endif
#ifndef BOOST_MATH_DISABLE_STD_FPCLASSIFY
  #define BOOST_MATH_DISABLE_STD_FPCLASSIFY 1
#endif
#ifndef BOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX
  #define BOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX 1
#endif
#ifndef CMSSW_GIT_HASH
  #define CMSSW_GIT_HASH "CMSSW_11_3_X_2021-03-22-2300"
#endif
#ifndef PROJECT_NAME
  #define PROJECT_NAME "CMSSW"
#endif
#ifndef PROJECT_VERSION
  #define PROJECT_VERSION "CMSSW_11_3_X_2021-03-22-2300"
#endif
#ifndef CMSSW_REFLEX_DICT
  #define CMSSW_REFLEX_DICT 1
#endif

#define _BACKWARD_BACKWARD_WARNING_H
// Inline headers
#include <utility>
#include <vector>

#include "DataFormats/Common/interface/Wrapper.h"
#include "SimDataFormats/JetMatching/interface/JetFlavourInfo.h"
#include "SimDataFormats/JetMatching/interface/JetFlavourInfoMatching.h"
#include "SimDataFormats/JetMatching/interface/JetFlavour.h"
#include "SimDataFormats/JetMatching/interface/JetFlavourMatching.h"
#include "SimDataFormats/JetMatching/interface/MatchedPartons.h"
#include "SimDataFormats/JetMatching/interface/JetMatchedPartons.h"

#undef  _BACKWARD_BACKWARD_WARNING_H


----- End Fatal Exception -------------------------------------------------
```
- [25202.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-77a5cf/13700/runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25+NANOUP15_PU25)
```
----- Begin Fatal Exception 23-Mar-2021 17:28:28 CET-----------------------
An exception of category 'FatalRootError' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 1
   [1] Running path 'nanoAOD_step'
   [2] Calling method for module SimpleCandidateFlatTableProducer/'fatJetMCTable'
   Additional Info:
      [a] Fatal Root Error: @SUB=TInterpreter::AutoParse
Error parsing payload code for class reco::JetFlavourInfo with content:

#line 1 "SimDataFormatsJetMatching_xr dictionary payload"

#ifndef CMS_DICT_IMPL
  #define CMS_DICT_IMPL 1
#endif
#ifndef _REENTRANT
  #define _REENTRANT 1
#endif
#ifndef GNUSOURCE
  #define GNUSOURCE 1
#endif
#ifndef __STRICT_ANSI__
  #define __STRICT_ANSI__ 1
#endif
#ifndef GNU_GCC
  #define GNU_GCC 1
#endif
#ifndef _GNU_SOURCE
  #define _GNU_SOURCE 1
#endif
#ifndef EIGEN_DONT_PARALLELIZE
  #define EIGEN_DONT_PARALLELIZE 1
#endif
#ifndef TBB_USE_GLIBCXX_VERSION
  #define TBB_USE_GLIBCXX_VERSION 90300
#endif
#ifndef TBB_SUPPRESS_DEPRECATED_MESSAGES
  #define TBB_SUPPRESS_DEPRECATED_MESSAGES 1
#endif
#ifndef TBB_PREVIEW_RESUMABLE_TASKS
  #define TBB_PREVIEW_RESUMABLE_TASKS 1
#endif
#ifndef BOOST_SPIRIT_THREADSAFE
  #define BOOST_SPIRIT_THREADSAFE 1
#endif
#ifndef PHOENIX_THREADSAFE
  #define PHOENIX_THREADSAFE 1
#endif
#ifndef BOOST_MATH_DISABLE_STD_FPCLASSIFY
  #define BOOST_MATH_DISABLE_STD_FPCLASSIFY 1
#endif
#ifndef BOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX
  #define BOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX 1
#endif
#ifndef CMSSW_GIT_HASH
  #define CMSSW_GIT_HASH "CMSSW_11_3_X_2021-03-22-2300"
#endif
#ifndef PROJECT_NAME
  #define PROJECT_NAME "CMSSW"
#endif
#ifndef PROJECT_VERSION
  #define PROJECT_VERSION "CMSSW_11_3_X_2021-03-22-2300"
#endif
#ifndef CMSSW_REFLEX_DICT
  #define CMSSW_REFLEX_DICT 1
#endif

#define _BACKWARD_BACKWARD_WARNING_H
// Inline headers
#include <utility>
#include <vector>

#include "DataFormats/Common/interface/Wrapper.h"
#include "SimDataFormats/JetMatching/interface/JetFlavourInfo.h"
#include "SimDataFormats/JetMatching/interface/JetFlavourInfoMatching.h"
#include "SimDataFormats/JetMatching/interface/JetFlavour.h"
#include "SimDataFormats/JetMatching/interface/JetFlavourMatching.h"
#include "SimDataFormats/JetMatching/interface/MatchedPartons.h"
#include "SimDataFormats/JetMatching/interface/JetMatchedPartons.h"

#undef  _BACKWARD_BACKWARD_WARNING_H


----- End Fatal Exception -------------------------------------------------
```
<details>
<summary>Expand to see more relval errors ...</summary>

- [10824.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-77a5cf/13700/runTheMatrix-results/10824.0_TTbar_13+2018+TTbar_13TeV_TuneCUETP8M1_GenSim+Digi+RecoFakeHLT+HARVESTFakeHLT+ALCA+Nano)
- [10224.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-77a5cf/13700/runTheMatrix-results/10224.0_TTbar_13+2017PU+TTbar_13TeV_TuneCUETP8M1_GenSim+DigiPU+RecoFakeHLTPU+HARVESTFakeHLTPU+Nano)
</details>



## RelVals-INPUT

- [1325.61](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-77a5cf/13700/runTheMatrixINPUT-results/1325.61_TTbar_13_106Xv1NanoAODINPUT+TTbar_13_106Xv1NanoAODINPUT+NANOAODMC2017_106XMiniAODv1)```1325.61_TTbar_13_106Xv1NanoAODINPUT+TTbar_13_106Xv1NanoAODINPUT+NANOAODMC2017_106XMiniAODv1/step2_TTbar_13_106Xv1NanoAODINPUT+TTbar_13_106Xv1NanoAODINPUT+NANOAODMC2017_106XMiniAODv1.log```
- [1325.81](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-77a5cf/13700/runTheMatrixINPUT-results/1325.81_TTbar_13_106Xv1NanoAODINPUT+TTbar_13_106Xv1NanoAODINPUT+NANOEDMMC2017_106XMiniAODv1+HARVESTNANOAODMC2017_106XMiniAODv1)```1325.81_TTbar_13_106Xv1NanoAODINPUT+TTbar_13_106Xv1NanoAODINPUT+NANOEDMMC2017_106XMiniAODv1+HARVESTNANOAODMC2017_106XMiniAODv1/step2_TTbar_13_106Xv1NanoAODINPUT+TTbar_13_106Xv1NanoAODINPUT+NANOEDMMC2017_106XMiniAODv1+HARVESTNANOAODMC2017_106XMiniAODv1.log```

silviodonato commented:
please test
the error was already in the IB

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-77a5cf/13738/summary.html
**COMMIT**: 0936dc591f392ba23dc05a9c9dcdf9f2ca386666
**CMSSW**: CMSSW_11_3_X_2021-03-23-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/33078/13738/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2639935
 - DQMHistoTests: Total failures: 478
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2639435
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 36 files compared)
 - Checked 155 log files, 37 edm output root files, 37 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_11_3_X_2021-03-23-2300+77a5cf/41862/triggerResults): found differences in 2 / 36 workflows

bachtis commented:
@rekovic seems this is ready?

panoskatsoulis commented:
Had a quick look at some of the comparison-plots as well and I think all look good - no unexpected changes in the WFs
I assume that minor changes in the PT distributions are because of the modified coefficients that lead to slightly different tracks and curvatures. (example here https://tinyurl.com/yzrwk4a5)

I think this is ready for the backport 
(we need to catch the deadline of the DQM so if also @rekovic and @silviodonato think that this is ready to go, 
please open the 11_2_X PR too)

bachtis commented:
@panoskatsoulis as you see Tyler has opened the backport

panoskatsoulis commented:
> @panoskatsoulis as you see Tyler has opened the backport

right my bad, mentioning this to the DQM
thx

silviodonato commented:
@cms-sw/l1-l2 do you have any comments?

cecilecaillol commented:
+l1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

qliphy commented:
+1
